lib/round-tracker.js

```diff
@@ -57,15 +57,22 @@ export async function startRoundTracker (pgPool) {
     }
   }
 
-  contract.on('RoundStart', (newRoundIndex) => {
+  const onRoundStart = (newRoundIndex) => {
     updateSparkRound(newRoundIndex).catch(err => {
       console.error('Cannot handle RoundStart:', err)
       Sentry.captureException(err)
     })
-  })
-
-  const currentRound = await updateSparkRound(await contract.currentRoundIndex())
-  return currentRound
+  }
+  contract.on('RoundStart', onRoundStart)
+  const close = () => contract.removeListener('RoundStart', onRoundStart)
+
+  try {
+    const currentRound = await updateSparkRound(await contract.currentRoundIndex())
+    return { ...currentRound, close }
+  } catch (err) {
+    close()
```

Comment:
When `await updateSparkRound()` throws, the caller does not have access to `close` and cannot unsubscribe.

To avoid leaking a background tracker, we must unsubscribe ourselves.

Since this is used by tests only, I think there is an alternative solution - remove all event listeners after each test. @juliangruber WDYT?

Reply:
What about we create `class RoundTracker`, on which you can call `start`, but also `close`? I believe we're outgrowing a single function here.